documentation/copyright_and_license_headers.md

```diff
@@ -0,0 +1,65 @@
+OP-TEE copyright and license headers
+====================================
+
+This document deals with the format of the copyright and license headers in
+OP-TEE source files. Such headers shall comply with the rules described here.
+
+New source files
+----------------
+
+- (Rule 1.1) Shall contain at least one copyright line
+- (Rule 1.2) Shall contain at least one SPDX license identifier
+- (Rule 1.3) Shall not contain the mention 'All rights reserved' or similar
+- (Rule 1.4) Copyrights and license identifiers shall appear in a comment block at
+  the first possible line in the file which can contain a comment.
+- (Rule 1.5) Files imported from external projects are not new files. The rules for
+  pre-existing files (below) apply.
+
+Example:
+```
+    /*
+     * Copyright (c) 2017, Linaro Limited
+     * SPDX-License-Identifier: BSD-2-Clause
+     */
+```
+
+Pre-existing or imported files
+------------------------------
+
+- (Rule 2.1) SPDX license identifiers shall be added according to the license
+  notice(s) in the file.
```

Comment:
Is there always an spdx id for an existing imported content? What if not?

Reply:
Not always. For example: https://github.com/OP-TEE/optee_os/blob/master/core/arch/arm/kernel/kern.ld.S#L28-L49. I could not find this exact license on the SPDX license page (https://spdx.org/licenses/).
In this case we can't do anything.

Reply:
So maybe replace `shall` by `should` in this doc.